pkg/backup/restorer/restorer_test.go

```diff
@@ -54,3 +54,85 @@ func TestRestorer(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, 0, c)
 }
+
+func TestRestorerWithExistingDatabase(t *testing.T) {
+	t.Parallel()
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// this is a test database that contains the following tables with one record each
+		// "a", "system_pending_tx", "system_id"
+		f, err := os.Open("testdata/database.db.zst")
+		require.NoError(t, err)
+		data, err := io.ReadAll(f)
+		require.NoError(t, err)
+		_, _ = w.Write(data)
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer ts.Close()
+
+	dirPath := t.TempDir()
+
+	// we create an existing database with the node id information stored
+	createExistingDatabase(t, dirPath)
+
+	databaseURL := fmt.Sprintf(
+		"file://%s?_busy_timeout=5000&_foreign_keys=on&_journal_mode=WAL",
+		path.Join(dirPath, "database.db"),
+	)
+	br, err := NewBackupRestorer(ts.URL, databaseURL)
+	require.NoError(t, err)
+	err = br.Restore()
+	require.NoError(t, err)
+
+	db, err := sql.Open("sqlite3", databaseURL)
+	require.NoError(t, err)
+
+	var a int
+	err = db.QueryRow("SELECT a FROM a LIMIT 1").Scan(&a)
+	require.NoError(t, err)
+	require.Equal(t, 1, a)
+
+	var address string
+	err = db.QueryRow("SELECT address FROM system_pending_tx").Scan(&address)
+	require.NoError(t, err)
+	require.Equal(t, "existing address", address)
+
+	var nodeID string
+	err = db.QueryRow("SELECT id FROM system_id").Scan(&nodeID)
+	require.NoError(t, err)
+	require.Equal(t, "existing node id", nodeID)
+}
+
+func createExistingDatabase(t *testing.T, dir string) {
```

Comment:
Create a database with `system_id` and `system_pending_tx` tables with one row each